autoload/neomake/makers/cargo.vim

```diff
@@ -1,11 +1,16 @@
 " vim: ts=4 sw=4 et
 
-function! neomake#makers#cargo#cargo()
+function! neomake#makers#cargo#cargo() abort
     return {
         \ 'args': ['build'],
         \ 'errorformat':
-            \   '%-Z%f:%l,' .
-            \   '%+C %s,' .
-            \   '%A%f:%l:%c: %*\d:%*\d\ %t%*[^:]: %m,',
+            \ '%Eerror[E%n]: %m,'.
+            \ '%Eerror: %m,'.
+            \ '%Wwarning: %m,'.
+            \ '%Inote: %m,'.
+            \ '%-Z\ %#-->\ %f:%l:%c,'.
```

Comment:
@fuine I believe `-Z` is deprecated, and it will trigger warnings. For example, using this Rust file:

```rust
fn main() {
    let $syntax_error = "hello";
}
```

The output in the quickfix window is:

```
|| the option `Z` is unstable and should only be used on the nightly compiler, but it is currently accepted for backwards compatibility; this will soon change, see issue #31847 for more details
/tmp/test.rs|2 col 9 error| expected pattern, found `$`
|| aborting due to 2 previous errors
```